core/src/main/java/com/graphhopper/storage/BaseGraph.java

```diff
@@ -1003,6 +1003,7 @@ public int getMaxId() {
         public boolean next() {
             while (true) {
                 edgeId++;
+                edgeRowCache = null;
```

Comment:
Not sure if this was intended or not, but when iterating over the edges, the edgeRowCache was not invalidated, which could lead to incorrect IntRefs.

Reply:
Nice finding. I think I have just done this in the normal iterator not in the "all iterator"